Assets/Scripts/Models/CharacterNameManager.cs

```diff
@@ -0,0 +1,69 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Character name manager that holds all the possible names and tryes to give a random and unused name everytime is requested.
+/// </summary>
+public class CharacterNameManager
+{
+    private static Queue<string> unusedNames;
+    private static Queue<string> usedNames;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CharacterNameManager"/> class.
+    /// </summary>
+    public CharacterNameManager()
+    {
+        unusedNames = new Queue<string>();
+        usedNames = new Queue<string>();
+    }
+
+    /// <summary>
+    /// Load new names to use for Characters and add them in a random way.
```

Comment:
Don't need the second half if you change the doc for `GetNewName` as suggested. Just "Load new names to use for Characters." The fact that you do the randomization here and not in `GetNewName` is an invisible implementation detail which may change without changing the behavior in any important way.